src/debugger/Debugger.cc

```diff
@@ -375,14 +375,24 @@ void Debugger::Cmd::disasm(std::span<const TclObject> tokens, TclObject& result,
 
 void Debugger::Cmd::disasmBlob(std::span<const TclObject> tokens, TclObject& result) const
 {
-	checkNumArgs(tokens, 4, Prefix{2}, "value addr");
+	checkNumArgs(tokens, Between{4, 5}, Prefix{2}, "value addr ?symbols?");
 	std::span<const uint8_t> bin = tokens[2].getBinary();
 	auto len = instructionLength(bin);
 	if (!len || *len > bin.size()) {
 		throw CommandException("Blob does not contain a complete Z80 instruction");
 	}
 	std::string dasmOutput;
-	dasm(bin.subspan(0, *len), word(tokens[3].getInt(getInterpreter())), dasmOutput);
+	unsigned addr = tokens[3].getInt(getInterpreter());
+	dasm(bin.subspan(0, *len), word(addr), dasmOutput,
+		[&](std::string& output, uint16_t a) {
+			TclObject key(a);
+			auto label = tokens[4].getDictValue(getInterpreter(), key).getString();
```

Comment:
Unconditionally using `tokens[4]` when the user only passes 4 (not 5) arguments may crash.